lib/swagger-module.ts

```diff
@@ -35,87 +45,107 @@ export class SwaggerModule {
+        swaggerAssetsAbsoluteFSPath,
+        { prefix: finalPath }
+      );
+    }
+  }
+
+  private static serveDocuments(
+    finalPath: string,
+    httpAdapter: HttpServer,
+    swaggerInitJS: string,
+    yamlDocument: string,
+    jsonDocument: string,
+    html: string
+  ) {
+    httpAdapter.get(`${finalPath}/swagger-ui-init.js`, (req, res) => {
+      res.type('application/javascript');
+      res.send(swaggerInitJS);
+    });
+
+    httpAdapter.get(finalPath, (req, res) => {
+      res.type('text/html');
+      res.send(html);
+    });
+
+    // fastify compatibility
+    httpAdapter.get(finalPath + '/', (req, res) => {
+      res.type('text/html');
+      res.send(html);
+    });
+
+    httpAdapter.get(`${finalPath}-json`, (req, res) => {
+      res.type('text/json');
+      res.send(jsonDocument);
+    });
+
+    httpAdapter.get(`${finalPath}-yaml`, (req, res) => {
+      res.type('text/yaml');
+      res.send(yamlDocument);
+    });
+  }
+
   public static setup(
     path: string,
     app: INestApplication,
     document: OpenAPIObject,
-    options?: SwaggerCustomOptions
+    options?: SwaggerCustomOptionsLegacy
   ) {
-    const httpAdapter = app.getHttpAdapter();
     const globalPrefix = getGlobalPrefix(app);
     const finalPath = validatePath(
       options?.useGlobalPrefix && globalPrefix && !globalPrefix.match(/^(\/?)$/)
         ? `${globalPrefix}${validatePath(path)}`
         : path
     );
-    if (httpAdapter && httpAdapter.getType() === 'fastify') {
-      return this.setupFastify(
-        finalPath,
-        httpAdapter,
-        document,
-        options as FastifySwaggerCustomOptions
-      );
-    }
-    return this.setupExpress(
-      finalPath,
-      app,
-      document,
-      options as ExpressSwaggerCustomOptions
-    );
-  }
 
-  private static setupExpress(
-    path: string,
-    app: INestApplication,
-    document: OpenAPIObject,
-    options?: ExpressSwaggerCustomOptions
-  ) {
+    // START: fastify backward compatibility layer
+    const { customOptions, extra } = processSwaggerOptions(options);
+    // END: fastify backward compatibility layer
+
+    const yamlDocument = jsyaml.dump(document);
+    const jsonDocument = JSON.stringify(document);
+    const html = buildSwaggerHTML(finalPath, document, customOptions);
+    const swaggerInitJS = buildSwaggerInitJS(document, customOptions);
+
     const httpAdapter = app.getHttpAdapter();
-    const swaggerUi = loadPackage('swagger-ui-express', 'SwaggerModule', () =>
-      require('swagger-ui-express')
-    );
-    const swaggerHtml = swaggerUi.generateHTML(document, options);
-    app.use(path, swaggerUi.serveFiles(document, options));
 
-    httpAdapter.get(path, (req, res) => res.send(swaggerHtml));
-    httpAdapter.get(path + '-json', (req, res) => res.json(document));
-  }
+    // START: fastify backward compatibility layer
+    const IS_FASTIFY = httpAdapter && httpAdapter.getType() === 'fastify';
```

Comment:
```suggestion
    const isFastify = httpAdapter && httpAdapter.getType() === 'fastify';
```
we use CONSTANT_CASE (`IS_FASTIFY`) for constants